Update feature flags docs (#1127)
mattermost/mattermost-developer-documentation: site/content/contribute/more-info/server/feature-flags.md

## Self-hosted releases

> For a feature-flagged feature to be included in a self-hosted release, the feature flag should be either removed or set to ``true`` in code. The feature flag can be set to ``true`` in code if we are not yet ready to fully remove the feature flag. Some [examples are here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L75).

Reviewer: ```suggestion
For a feature-flagged feature to be included in a self-hosted release, the feature flag should be either removed or set to ``true`` in code. The feature flag shouldn't be set to ``true`` in code if we're not yet ready to fully remove the feature flag. Some [examples are here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L75).
```

## Self-hosted releases

> For a feature-flagged feature to be included in a self-hosted release, the feature flag should be either removed or set to ``true`` in code. The feature flag shouldn't be set to ``true`` in code if we're not yet ready to fully remove the feature flag. Some [examples are here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L75).

Reviewer 2: > The feature flag shouldn't be set to ``true`` in code if we're not yet ready to fully remove the feature flag.

I'm not understanding because in the previous statement we said it was perfectly OK to release a feature flag in self-hosted with a `true` value.

Author: Clarified the text 👍

## Tests

Tests should be written to verify all states of the feature flag. Tests should cover any migrations that may take place in both directions (i.e., from "off" to "on" and from "on" to "off"). Ideally E2E tests should be written before the feature is merged, or at least before the feature flag is removed.

Features that are not yet ready to be shipped in a self-hosted release are automatically disabled if the flag is set to ``false`` in code.

## Examples of feature flags

Some [examples are here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L75).

## FAQ

1. What is the expected default value for boolean feature flags? Is it `true` or `false`?

>    - Initially `false`. The idea is to use them to slowly roll out a feature. When the code is deployed, the feature flag is not enabled yet. See more details on feature flag rollout timelines [here]({{< ref "/contribute/server/feature-flags#timelines-for-rollouts" >}}).

Reviewer 3: ```suggestion
   - Initially `false`. The idea is to use them to slowly roll out a feature. When the code is deployed, the feature flag is not enabled yet. See more details on feature flag rollout timelines [here]({{< ref "/contribute/more-info/server/feature-flags#timelines-for-rollouts" >}}).
```

>    - Initially `false`. The idea is to use them to slowly roll out a feature. When the code is deployed, the feature flag is not enabled yet. See more details on feature flag rollout timelines [here]({{< ref "/contribute/more-info/server/feature-flags#timelines-for-rollouts" >}}).

Reviewer 2: I would agree that for cloud release this is correct: a feature flag should default to `false`. That gives us a week to vet that the feature doesn't break things.

However, for a self-hosted releases I think it's unreasonable to expect all feature flags to default to `false`.

Author: Clarified the text 👍
